### ISSUE TITLE
Neon compilation bug

### ISSUE THREAD

**Issue by ojwoodford** [QuantStack/xsimd#293]
There are bugs in several of your load_unaligned/store_unaligned function implementations. They call themselves, instead of the aligned version, causing an infinite recursion.

```
xsimd/types/xsimd_neon_float.hpp:274:5: error: all paths through this function will call itself [-Werror,-Winfinite-recursion]
xsimd/types/xsimd_neon_float.hpp:300:5: error: all paths through this function will call itself [-Werror,-Winfinite-recursion]
xsimd/types/xsimd_neon_int64.hpp:178:5: error: all paths through this function will call itself [-Werror,-Winfinite-recursion]
xsimd/types/xsimd_neon_uint64.hpp:181:5: error: all paths through this function will call itself [-Werror,-Winfinite-recursion]
xsimd/types/xsimd_neon_complex.hpp:157:5: error: all paths through this function will call itself [-Werror,-Winfinite-recursion]
```

**Comment by ojwoodford** [comment]
This also suggests to me that your build tests don't catch this as an error, when they probably should.

**Comment by wolfv** [comment]
indeed, thanks! they should call `store_unaligned`...

**Comment by JohanMabille** [comment]
Fixed in #294
